src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13199,7 +13199,9 @@ private AssignmentExpressionSyntax ParseObjectInitializerNamedAssignment()
             return _syntaxFactory.AssignmentExpression(
                 SyntaxKind.SimpleAssignmentExpression,
                 this.ParseIdentifierName(),
-                this.EatToken(SyntaxKind.EqualsToken),
+                this.CurrentToken.Kind == SyntaxKind.ColonToken
+                    ? this.EatTokenWithPrejudice(SyntaxKind.EqualsToken)
```

Comment:
```suggestion
                    ? this.EatTokenAsKind(SyntaxKind.EqualsToken)
```

Your branch seems to be out of date.